scalafix-diff/shared/src/main/scala/scalafix/diff/GitDiffParser.scala

```diff
@@ -0,0 +1,193 @@
+package scalafix.diff
+
+import scala.util.matching.Regex
+
+import java.nio.file.Path
+
+case class Range(start: Int, length: Int)
+
+sealed trait GitDiff
+case class NewFile(path: Path) extends GitDiff
+case class ModifiedFile(path: Path, changes: List[Range]) extends GitDiff
+
+private[diff] case class Hunk(
+    originalLine: Int,
+    originalCount: Int,
+    revisedLine: Int,
+    revisedCount: Int)
+
+private[diff] object HunkExtractor {
```

Comment:
Can this be `Hunk` without "Extractor"?

Reply:
Can we move this to `internal.diff`? The main motivation is simplified mima maintenance, you will need to write a lot of filters for every new/removed field in `private[diff]` classes.

Reply:
Can we also split Hunk/GitDiffParser/PeekableSource into different files.